src/Microsoft.PowerShell.Commands.Utility/commands/utility/Select-Object.cs

```diff
@@ -648,16 +648,22 @@ private void FilteredWriteObject(PSObject obj, List<PSNoteProperty> addedNotePro
 
                 if (isObjUnique)
                 {
-                    SetPSCustomObject(obj);
+                    SetPSCustomObject(obj, newPSObject: addedNoteProperties.Count > 0);
                     _uniques.Add(new UniquePSObjectHelper(obj, addedNoteProperties.Count));
                 }
             }
         }
 
-        private void SetPSCustomObject(PSObject psObj)
+        private void SetPSCustomObject(PSObject psObj, bool newPSObject)
```

Comment:
Is this optimization really needed?  Seems like just checking if the typename is already there is sufficient, simple, and not a big impact.

Reply:
Current performance for the scenario is extremely fast. If we add the type check the scenario performance will greatly decrease. So I added the simple optimization.

Reply:
Is there a real world case where the Selected.* typename is not in index 0?  Alternatively, we could just check that.

Reply:
_Assumptions_ in this regard are not very reliable.
The fix logic is as simple as if object is not changed then pass throughout otherwise check Selected.* typename presents.